### PR TITLE
feature: rspec integration

### PR DESCRIPTION
**Pull request by jirikuncar**
Adds integration for [RSpec](https://rspec.info/).

### CODE REVIEW

File: lib/ddtrace/contrib/rspec/ext.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      # RSpec integration constants
+      module Ext
+        APP = 'rspec'.freeze
+        ENV_ANALYTICS_ENABLED = 'DD_TRACE_RSPEC_ANALYTICS_ENABLED'.freeze
+        ENV_ANALYTICS_SAMPLE_RATE = 'DD_TRACE_RSPEC_ANALYTICS_SAMPLE_RATE'.freeze
+        ENV_ENABLED = 'DD_TRACE_RSPEC_ENABLED'.freeze
+        ENV_OPERATION_NAME = 'DD_TRACE_RSPEC_OPERATION_NAME'.freeze
+        FRAMEWORK = 'rspec'.freeze
+        OPERATION_NAME = 'rspec.it'.freeze
+        EXAMPLE_GROUP_OPERATION_NAME = 'rspec.example'.freeze
```
Comment (marcotc) [inline lib/ddtrace/contrib/rspec/ext.rb:13]: To match rspec nomenclature, I suggest we name these:
```suggestion
        OPERATION_NAME = 'rspec.example'.freeze
        EXAMPLE_GROUP_OPERATION_NAME = 'rspec.example_group'.freeze
```

These would follow rspec's two main structures:

* example: https://github.com/rspec/rspec-core/blob/5d87c28bb828ba14b54a0d8cfddf044894009314/lib/rspec/core/example_group.rb#L145
* example_group: https://github.com/rspec/rspec-core/blob/5d87c28bb828ba14b54a0d8cfddf044894009314/lib/rspec/core/example_group.rb#L246

There's also a top-level concept called "nested_shared_group", but I'm not sure these will look different from an example group at runtime: https://github.com/rspec/rspec-core/blob/5d87c28bb828ba14b54a0d8cfddf044894009314/lib/rspec/core/example_group.rb#L317

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -1499,6 +1499,31 @@ Where `options` is an optional `Hash` that accepts the following parameters:
...
+```
+
+Where `options` is an optional `Hash` that accepts the following parameters:
+
+| Key | Description | Default |
+| --- | ----------- | ------- |
+| `analytics_enabled` | Enable analytics for spans produced by this integration. `true` for on, `nil` to defer to global setting, `false` for off. | `true` |
+| `enabled` | Defines whether RSpec tests should be traced. Useful for temporarily disabling tracing. `true` or `false` | `true` |
+| `service_name` | Service name used for `rspec` instrumentation. | `'rspec'` |
+| `operation_name` | Operation name used for `rspec` instrumentation. Useful if you want rename automatic trace metrics e.g. `trace.#{operation_name}.errors`. | `'rspec.it'` |
```
Comment (jirikuncar) [inline docs/GettingStarted.md:1525]:
```suggestion
| `operation_name` | Operation name used for `rspec` instrumentation. Useful if you want rename automatic trace metrics e.g. `trace.#{operation_name}.errors`. | `'rspec.example'` |
```

**Review by marcotc — COMMENTED** [review]
Anything you think is still missing, @jirikuncar? The PR as it is looks good.

**Comment by marcotc** [comment]
Feel free to promote it from draft if you are comfortable.

**Review by marcotc — APPROVED** [review]
🎉

**Review by ericmustin — APPROVED** [review]
👍

**Pull request merged**